modules/database/src/adapters/mongoose-adapter/index.ts

```diff
@@ -382,6 +382,10 @@ export class MongooseAdapter extends DatabaseAdapter<MongooseSchema> {
     return result;
   }
 
+  async syncSchema(name: string) {
```

Comment:
this needs to be supported in mongoose as well and we should $unset all fields that have been removed